install/migrations/update_10.0.5_to_10.0.6/unmanaged.php

```diff
@@ -0,0 +1,230 @@
+            'pattern'   => 1
+        ]
+    ],
+    'action'    => [
+        [
+            'field'         => '_inventory',
+            'action_type'  => "assign",
+            'value'         => RuleImportAsset::RULE_ACTION_LINK_OR_IMPORT,
+        ]
+    ]
+];
+
+$rules[] = [
+    'name'      => 'Unmanaged import (by name)',
+    'uuid'      => 'glpi_rule_import_asset_unmanaged_import_name',
+    'match'     => 'AND',
+    'is_active' => 1,
+    'criteria'  => [
+        [
+            'criteria'  => 'itemtype',
+            'condition' => Rule::PATTERN_IS,
+            'pattern'   => 'Unmanaged'
+        ],
+        [
+            'criteria'  => 'name',
+            'condition' => Rule::PATTERN_EXISTS,
+            'pattern'   => 1
+        ]
+    ],
+    'action'    => [
+        [
+            'field'         => '_inventory',
+            'action_type'  => "assign",
+            'value'         => RuleImportAsset::RULE_ACTION_LINK_OR_IMPORT,
+        ]
+    ]
+];
+
+$rules[] = [
+    'name'      => 'Unmanaged import denied',
+    'uuid'      => 'glpi_rule_import_asset_unmanaged_import_denied',
+    'match'     => 'AND',
+    'is_active' => 1,
+    'criteria'  => [
+        [
+            'criteria'  => 'itemtype',
+            'condition' => Rule::PATTERN_IS,
+            'pattern'   => 'Unmanaged'
+        ]
+    ],
+    'action'    => [
+        [
+            'field'         => '_inventory',
+            'action_type'  => "assign",
+            'value'         => RuleImportAsset::RULE_ACTION_DENIED
+        ]
+    ]
+];
+
+$query = "SELECT MAX(`ranking`) FROM `glpi_rules`";
+$result = $DB->query($query);
+$ranking   = $DB->result($result, 0, 0);
+
+foreach ($rules as $rule) {
+    $rulecollection = new RuleImportAssetCollection();
+    $input = [
+        'is_active' => $rule['is_active'],
+        'name'      => $rule['name'],
+        'uuid'      => $rule['uuid'],
+        'match'     => $rule['match'],
+        'sub_type'  => RuleImportAsset::getType(),
+        'ranking'   => $ranking
+    ];
+
+    if ($rulecollection->getFromDBByCrit(['uuid' => $rule['uuid']])) {
+        //rule already exists with uuid, ignore.
+        continue;
+    }
+
+    //create rule
+    $stmt = $DB->prepare($DB->buildInsert(
+        "glpi_rules",
+        [
+            'is_active' => $input['is_active'],
+            'name'      => $input['name'],
+            'match'     => $input['match'],
+            'sub_type'  => $input['sub_type'],
+            'uuid'      => $input['uuid'],
+            'ranking'   => $input['ranking'],
+        ]
+    ));
+
+    if (false === $stmt) {
+        $msg = "Error preparing statement in table glpi_rules";
+        throw new \RuntimeException($msg);
+    }
+
+    $res = $stmt->execute();
+    if (false === $res) {
+        $msg = $stmt->error;
+        $msg .= "\nError execution statement in table glpi_rules\n";
+        $msg .= print_r($row, true);
+        throw new \RuntimeException($msg);
+    }
```

Comment:
If you wan tot use a prepared statement, you have to use prepared parameters in initial query an dbind them later. Or you can use direct query rather than a prepared statement.